mantidproject/mantid
# Testable Python Documents

**Description of work.**

This PR migrates the Python training docs to Sphinx and makes many of the code blocks testable.

This is an updated version of #23724, which had gone stale.

**To test:**


* Build the `docs-html` target.
* Open `html/index.html`
* Navigate to `Training` then `Introduction to Python`
* Check that the pages have built and links work
* Test some of the examples to see if they work


Fixes #xxxx. 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
